src/eduid_userdb/proofing/user.py

```diff
@@ -0,0 +1,153 @@
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+
+import bson
+import copy
+import datetime
+
+from .proofing_element import NinProofingElement, ProofingLetterElement
+from eduid_userdb.exceptions import UserHasUnknownData
+
+__author__ = 'lundberg'
+
+
+class ProofingUser(object):
+    def __init__(self, data, raise_on_unknown=True):
+        self._data_in = copy.deepcopy(data)  # to not modify callers data
+        self._data = dict()
+
+        # things without setters
+        # _id
+        _id = self._data_in.pop('_id', None)
+        if _id is None:
+            _id = bson.ObjectId()
+        if not isinstance(_id, bson.ObjectId):
+            _id = bson.ObjectId(_id)
+        self._data['_id'] = _id
+        # user_id
+        user_id = self._data_in.pop('user_id')
+        if not isinstance(_id, bson.ObjectId):
+            user_id = bson.ObjectId(_id)
+        self._data['user_id'] = user_id
+
+        self.modified_ts = self._data_in.pop('modified_ts', None)
+
+        if len(self._data_in) > 0:
+            if raise_on_unknown:
+                raise UserHasUnknownData('User {!s} unknown data: {!r}'.format(
+                    self.user_id, self._data_in.keys()
+                ))
+            # Just keep everything that is left as-is
+            self._data.update(self._data_in)
+
+    def __repr__(self):
+        return '<eduID {!s}: {!s}>'.format(self.__class__.__name__, self.user_id)
+
+    @property
+    def user_id(self):
+        """
+        Get the user's oid in MongoDB.
+
+        :rtype: bson.ObjectId
+        """
+        return self._data['user_id']
+
+    @property
+    def modified_ts(self):
+        """
+        :return: Timestamp of last modification in the database.
+                 None if User has never been written to the database.
+        :rtype: datetime.datetime | None
+        """
+        return self._data.get('modified_ts')
+
+    @modified_ts.setter
+    def modified_ts(self, value):
+        """
+        :param value: Timestamp of modification.
+                      Value None is ignored, True is short for datetime.utcnow().
+        :type value: datetime.datetime | True | None
+        """
+        if value is None:
+            return
+        if value is True:
+            value = datetime.datetime.utcnow()
+        self._data['modified_ts'] = value
+
+
+
+class NinProofingUser(ProofingUser):
```

Comment:
As discussed AFK, I would prefer to avoid the intermediate NinProofingUser for now. Collapse into class LetterProofingUser(ProofingUser).